polyglot.js

```diff
@@ -362,18 +380,86 @@ class PolyGlot {
         }
         sheet._polyglotEditor = true;
     }
-    renderJournalSheet(journalSheet) {
+
+    renderJournalSheet(journalSheet,html) {
         this._addPolyglotEditor(journalSheet);
-        if (journalSheet.entity.owner) return;
-        if (game.user.isGM && !game.settings.get("polyglot", "runifyGM")) return;
+        if (journalSheet.entity.owner || game.user.isGM) {
+            let runes = false;
+            let texts = [];
+            let openBtn = $(`<a class="polyglot-button" title="Toggle Runes"><i class="fas fa-unlink"></i> Runes</a>`);
+            openBtn.click(ev => {
+                let button = html.closest('.app').find('.polyglot-button')[0].firstChild
+                runes = !runes
+                button.className = runes === false ? 'fas fa-unlink' : 'fas fa-link';
+                if (runes) {
+                    const spans = journalSheet.element.find("span.polyglot-journal");
+                    for (let span of spans.toArray()) {
+                        const lang = span.dataset.language;
+                        if (!lang) continue;
+                        texts.push(span.textContent)
+                        span.textContent = this.generateRune(span.textContent,game.settings.get('polyglot','salt') === true ? journalSheet._id : lang)
+                        span.style = this.tongues[lang] === undefined ? 'font:' + this.alphabets[this.tongues._default] : 'font:' + this.alphabets[this.tongues[lang]]
+                    }
+                }
+                else {
+                    const spans = journalSheet.element.find("span.polyglot-journal");
+                    var i = 0;
+                    for (let span of spans.toArray()) {
+                        const lang = span.dataset.language;
+                        if (!lang) continue;
+                        span.textContent = texts[i]
```

Comment:
I think this will break if you do this : 
- have 3 texts in a language
- unscramble them (`texts` is now an array of 3)
- add a 4th text in between the two
- scramble them (you try to map the wrong index)

I *think* though that you can't edit a journal without it getting re-rendered, so that should never happen? can you confirm ?

Reply:
Yes it gets re-rendered. Thats why the code removes the last button and adds a new one, and why the button always initialize on 'unlink' mode.

Reply:
Cool, thanks for confirming!